e2e_test/js-tests/test_viem_smoketest.mjs

```diff
@@ -0,0 +1,170 @@
+	switch (process.env.NETWORK) {
+		case 'alfajores':
+			return celoAlfajores
+		default:
+			return devChain
+	};
+})();
+
+// Set up clients/wallet
+const publicClient = createPublicClient({
+	chain: chain,
+	transport: http(),
+});
+const account = privateKeyToAccount(process.env.ACC_PRIVKEY);
+const walletClient = createWalletClient({
+	account,
+	chain: chain,
+	transport: http(),
+});
+
+// check checks that the receipt has status success and that the transaction
+// type matches the expected type, since viem sometimes mangles the type when
+// building txs.
+async function check(txHash, type) {
+	const receipt = await publicClient.waitForTransactionReceipt({ hash: txHash });
+	const transaction = await publicClient.getTransaction({ hash: txHash });
+	assert.equal(transaction.type, type, "transaction type does not match");
+	assert.equal(receipt.status, "success", "receipt status 'failure'");
+}
+
+// sendTypedTransaction sends a transaction with the given type and an optional
+// feeCurrency.
+async function sendTypedTransaction(type, feeCurrency){
+		 return await walletClient.sendTransaction({
+			to: "0x00000000000000000000000000000000DeaDBeef",
+			value: 1,
+			type: type,
+			feeCurrency: feeCurrency,
+		});
+}
+
+// sendTypedCreateTransaction sends a create transaction with the given type
+// and an optional feeCurrency.
+async function sendTypedCreateTransaction(type, feeCurrency){
+		 return await walletClient.deployContract({
+			type: type,
+			feeCurrency: feeCurrency,
+			bytecode: testContractJSON.bytecode.object,
+			abi: testContractJSON.abi,
+			 // The constructor args for the test contract at ../debug-fee-currency/DebugFeeCurrency.sol
+			args:[1n, true, true, true],
+		});
+}
+
+// verifyTypedTransactions is a helper function that submits a send and create
+// transaction of the given type with optional feeCurrency and checks the
+// results.
+async function verifyTypedTransactions(type, feeCurrency){
+		const send = await sendTypedTransaction(type, feeCurrency);
+		await check(send, type);
+		const create = await sendTypedCreateTransaction(type, feeCurrency);
+		await check(create, type);
+}
+
+describe("viem smoke test", () => {
+	it("send legacy tx", async () => {
+		const type = "legacy";
+		await verifyTypedTransactions(type);
+	}).timeout(30_000);
+
+	it("send eip2930 tx", async () => {
+		const type = "eip2930";
+		await verifyTypedTransactions(type);
+	}).timeout(10_000);
+
+	it("send eip1559 tx", async () => {
+		const type = "eip1559";
+		await verifyTypedTransactions(type);
+	}).timeout(10_000);
+
+	it("send cip64 tx", async () => {
+		const type = "cip64";
+		await verifyTypedTransactions(type, process.env.FEE_CURRENCY);
+	}).timeout(10_000);
+
+	// This test is failing because the produced transaction is of type cip64.
+	// I guess this is a problem with the viem internals.
+	it.skip("cip42 not supported", async () => {
+		const type = "cip42";
+		await verifyTypedTransactions(type);
+	}).timeout(10_000);
```

Comment:
Interesting and I think your explanation is correct. We could try running the same test against Celo L1 to make sure it's not a change in the chain behaviour that is causing this.

Reply:
@piersy just to get the explanation right - viem with the new l2 alfajores definitions does not seem to know about "cip42" types and instead produces a cip64 transaction, that then fails the assertion? 

I would classify this as a viem bug, since it shouldn't even allow the deprecated `cip42' type and fail early on unknown types (at least in my expectation).

Reply:
yeah, I'd classify it as a viem bug, i will raise it with the dev tooling team

Reply:
However viem still needs to be able to send cip42 txs to the celo l1, so it does still need that functionality.

Reply:
Does it? It could just send CIP-64 without missing any features.

Reply:
Unless people really want to set the gatewayFee and gatewayFeeRecipient! I'll check with the devtooling team to see what the expected approach for viem is.

Reply:
You can't set those to non-zero values anymore (celo-blockchain will reject the tx otherwise), so nothing is lost.

Reply:
Ok, was unaware!

Reply:
@karlb @ezdac I've moved this test into the smoketest_unsupported_txs golang test. Where we can be sure that we are sending an invalid transaction, and therefore ensure that the server correctly rejects it.